### PR TITLE
[W5.11][W11-A2] Loh Cai Jun

### PR DESCRIPTION
**Pull request by caijun7**
Implement new enhancement 2 - Added Sort Command to sort all entries in the address book by name in alphabetical order

### CODE REVIEW

**Comment by dannyngmx94** [comment]
Good job with the new feature. I think that you could commit more often on the minor changes rather than a massive change. This will be helpful when you try to revert to a previous commit.

File: test/expected.txt [hunk 1]
```diff
@@ -297,6 +299,32 @@
...
+|| 	2. Betsy Choo Tags: [secretive]
+|| 	3. Dickson Ee Phone: 444444 Address: 444, delta street Tags: [friends]
+|| 
+|| 3 persons listed!
+|| ===================================================
+|| Enter command: || [Command entered:  sort]
+|| Address book has been sorted!
+|| ===================================================
+|| Enter command: || [Command entered:  list]
+|| 	1. Betsy Choo Tags: [secretive]
```
Comment (okkhoy) [inline test/expected.txt:322]: :+1:  nicely done test! Shows clearly the before and after :clap:

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.*;
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:13]: Don't import everything using *; Import only the (necessary) classes explicitly.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+     * @param expectedList
+     */
+    private void assertSortSuccessful(List<Person> expectedList) {
+        sortCommand = new SortCommand();
+        sortCommand.setData(addressBook, Collections.emptyList());
+        sortCommand.execute();
+
+        UniquePersonList people = addressBook.getAllPersons();
+        UniquePersonList expected = null;
+        try {
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:61]: In JUnit tests, you can simply throw the exception, if any; you can skip the `try ... catch`

**Review by okkhoy — COMMENTED** [review]
Good job on updating the JUnit tests as well. Neat PR again :+1:

**Comment by okkhoy** [comment]
@Caijun7 
Some comments added. Please ack & close the PR after reading comments.

**Pull request closed without merging**